react/features/invite/components/InviteDialog.js

```diff
@@ -0,0 +1,107 @@
+import { translate } from '../../base/i18n';
+import JitsiMeetJS from '../../base/lib-jitsi-meet';
+import {
+    getLocalParticipant,
+    PARTICIPANT_ROLE
+} from '../../base/participants';
+
+import PasswordContainer from './PasswordContainer';
+import ShareLinkForm from './ShareLinkForm';
+
+/**
+ * A React Component for displaying other components responsible for copying the
+ * current conference url and for setting or removing a conference password.
+ */
+class InviteDialog extends Component {
+    /**
+     * InviteDialog component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        /**
+         * The redux store representation of the JitsiConference.
+         *
+         */
+        _conference: React.PropTypes.object,
+
+        /**
+         * The redux store representation of the local user.
+         */
+        _localParticipant: React.PropTypes.object,
+
+        /**
+         * The url for the JitsiConference.
+         */
+        conferenceUrl: React.PropTypes.string,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Reports an analytics event for the invite modal being closed.
+     *
+     * @inheritdoc
+     */
+    componentWillUnmount() {
+        JitsiMeetJS.analytics.sendEvent('toolbar.invite.close');
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        const isModerator
+            = this.props._localParticipant.role === PARTICIPANT_ROLE.MODERATOR;
+
+        return (
+            <Dialog
+                cancelDisabled = { true }
+                okTitleKey = 'dialog.done'
+                titleString = { this.props.t(
+                    'invite.inviteTo',
+                    { conferenceName: this.props._conference.room }) } >
+                <div className = 'invite-dialog'>
+                    <ShareLinkForm toCopy = { this.props.conferenceUrl } />
+                    <PasswordContainer
+                        conference = { this.props._conference.conference }
+                        locked = { this.props._conference.locked }
+                        lockedLocally
+                            = { this.props._conference.lockedLocally }
+                        password = { this.props._conference.password }
+                        showPasswordEdit = { isModerator } />
+                </div>
+            </Dialog>
+        );
+    }
+}
+
+/**
+ * Maps (parts of) the Redux state to the associated InviteDialog's props.
+ *
+ * @param {Object} state - The Redux state.
+ * @private
+ * @returns {{
+ *     _conference: Object,
+ *     _localParticipant: Object
+ * }}
+ */
+function _mapStateToProps(state) {
+    return {
+        _conference: state['features/base/conference'],
+        _localParticipant:
+            getLocalParticipant(state['features/base/participants'])
```

Comment:
I'm not sure why you're mapping the whole feature states into props.  I guess you could see how it makes sense for `_conference` (or maybe not), I don't see any sense in `_localParticipant`.